addOns/scripts/src/main/java/org/zaproxy/zap/extension/scripts/CommandPanel.java

```diff
@@ -94,10 +99,86 @@ private SyntaxHighlightTextArea getTxtOutput() {
             if (listener != null) {
                 this.syntaxTxtArea.addKeyListener(listener);
             }
+            if (isDarkLaF()) {
+                darkLaF = true;
+                try {
+                    Theme theme =
+                            Theme.load(
+                                    HttpPanelSyntaxHighlightTextArea.class.getResourceAsStream(
+                                            "/org/fife/ui/rsyntaxtextarea/themes/dark.xml"));
+                    theme.apply(this.syntaxTxtArea);
+                } catch (IOException e) {
+                    // Ignore
+                }
+            } else {
+                darkLaF = false;
+            }
         }
         return this.syntaxTxtArea;
     }
 
+    private Boolean supportsDarkLaF;
+    private Method isDarkLookAndFeelMethod;
+    private boolean darkLaF;
+
+    private boolean isDarkLaF() {
+        // TODO Update to calling the DisplayUtils.isDarkLookAndFeel() method directly once it is
+        // available
+        if (supportsDarkLaF == null) {
+            supportsDarkLaF = false;
+            try {
+                Class<DisplayUtils> cls = DisplayUtils.class;
+                isDarkLookAndFeelMethod = cls.getMethod("isDarkLookAndFeel");
+                if (isDarkLookAndFeelMethod != null) {
+                    supportsDarkLaF = true;
+                }
+            } catch (Exception e) {
+                // Ignore
+            }
+        }
+        if (isDarkLookAndFeelMethod != null) {
+            try {
+                Object obj = isDarkLookAndFeelMethod.invoke(null);
+                if (obj instanceof Boolean) {
+                    return (Boolean) obj;
+                }
+            } catch (Exception e) {
+                // Ignore
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public void updateUI() {
+        super.updateUI();
+        if (darkLaF != isDarkLaF()) {
+            if (isDarkLaF()) {
+                try {
+                    Theme theme =
+                            Theme.load(
+                                    HttpPanelSyntaxHighlightTextArea.class.getResourceAsStream(
+                                            "/org/fife/ui/rsyntaxtextarea/themes/dark.xml"));
+                    theme.apply(getTxtOutput());
+                } catch (IOException e) {
+                    // Ignore
+                }
+                darkLaF = true;
+            } else {
+                try {
+                    Theme theme =
+                            Theme.load(
+                                    HttpPanelSyntaxHighlightTextArea.class.getResourceAsStream(
+                                            "/org/fife/ui/rsyntaxtextarea/themes/default.xml"));
+                    theme.apply(getTxtOutput());
+                } catch (IOException e) {
+                    // Ignore
+                }
```

Comment:
Since this same code appears 3 times with the only difference being the object it's being applied to maybe just extract a method?

Reply:
Maybe also extract the xml file path to a constant?